packages/angular_devkit/schematics/src/tree/interface.ts

```diff
@@ -91,6 +91,16 @@ export interface Tree {
    * @returns A string containing the contents of the file; if not found, returns `null`.
    */
   readText(path: string): string | null;
+
+  /**
+   * Reads and parses a file from the Tree as a UTF-8 encoded JSON file.
+   * Parsing supports JSON comments and trailing commas.
+   *
+   * @param path The path of the file to read.
+   * @returns A JsonValue containing the parsed contents of the file; if not found, returns `undefined`.
+   */
+  readJSON(path: string): JsonValue | undefined;
```

Comment:
If this supports comments and trailing commas, should we call it `readJSONC()` or something similar? It will successfully parse inputs that aren't JSON, which makes me think that `readJSON()` is the wrong name.

Reply:
I think that's a distinction that may lead to more confusion than benefit in this case.  Most `.json` files that would be typically interacted with in a schematic (tsconfig.json/angular.json/eslint.json/tslint.json/etc.) support comments (package.json withstanding) but are still generally considered to be JSON files.  From looking through TypeScript's JSON parsing functionality, it also does not appear to mention JSONC either but it does support comments.
But if there is a demand for strict adherence to the JSON spec, we could always add a `strict` option to this function.

Reply:
I agree that many of these files are really JSONC despite having a `.json` extension. I think if someone wants to parse strict JSON, they can just do `JSON.parse(tree.readText())`, so I don't see much motivation to support that. I see this function as supporting the JSONC use case, which is more complicated to implement and requires an extra dependency. Does it make sense to label it as such if that's all we want to support here?

Reply:
If JSONC was formalized then i think it would make sense.  However, it's more of a shorthand for "JSON with comments" but without formalization as to the type/format of comments.  The JSON specification (RFC 8259 section 9) does allow for a compliant parser to support extensions to the specification though.  So i think `readJson` can be interpreted as parsing commonly used variants of JSON.  The documentation comment has been updated to make the parsing extensions more explicit.

Reply:
TIL the JSON spec allows extensions. I guess that means JSON is a superset of itself?

Based on that reading, I agree that `readJson()` is an accurate name (though I don't know how many people would actually interpret it correctly). As long as the documentation explains what extensions this includes, then I think we're ok.